### PR TITLE
Issue 54: suppress archiving if location is set to None

### LINKED ISSUE #54: Allow suppression of "archive" JSON file creation in `ClientSettings`
### Checklist

- [X] There are [no similar issues or pull requests](https://github.com/Midnighter/structurizr-python/issues) for this yet.

### Is your feature related to a problem? Please describe it.

Currently, whenever you fetch a workspace from the Structurizr API, a file is created with the zipped JSON.  If you don't desire this behaviour then you can work around it (e.g. by using `tempfile.TemporaryDirectory()` and putting that in the `ClientSettings` then cleaning it up afterwards) but this is a bit clumsy and runs the risk of leaving stray files/directories if the process is terminated prematurely.

## Describe the solution you would like.

Currently `ClientSettings` defaults `workspace_archive_location` to `Path.cwd()` if unspecified.  I propose that instead it should be made optional and if `None` then `StructurizrClient.get_workspace()` not generate an archive file, plus by default it is `None` so archive files do not get created.

## Describe alternatives you considered

Alternatively, we could add an explicit boolean flag to `ClientSettings` to indicate whether archive files should be created, but this seems a little unpythonic.

### PR DESCRIPTION
**Pull request by yt-ms**
* [X] fix https://github.com/Midnighter/structurizr-python/issues/54
* [X] description of feature/fix
* [X] tests added/passed
* [X] add an entry to the [next release](../CHANGELOG.rst)

If `archive_workspace_location` is explicitly set to `None` in the client settings, then this will suppress the client creating archived zip files of each workspace it downloads from the API.

------------

THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE Apache License v.2.0. YOU MAY OBTAIN A COPY OF THE LICENSE AT https://www.apache.org/licenses/LICENSE-2.0.

THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

### CODE REVIEW

File: src/structurizr/api/structurizr_client.py [hunk 1]
```diff
@@ -276,13 +276,14 @@ def _add_headers(
...
     def _archive_workspace(self, json: str) -> None:
         """Store the workspace."""
-        location = self._create_archive_filename()
-        logger.debug(
-            f"Archiving workspace {self.workspace_id} to"
-            f" '{self.workspace_archive_location}'."
-        )
-        with gzip.open(location, mode="wt") as handle:
-            handle.write(json)
+        if self.workspace_archive_location is not None:
```
Comment (Midnighter) [inline src/structurizr/api/structurizr_client.py]: I think it's simpler to just exit early:

```suggestion
        if self.workspace_archive_location is None:
            return
```

Comment (Midnighter) [reply src/structurizr/api/structurizr_client.py]: And then adjust indentation level.

**Review by Midnighter — CHANGES_REQUESTED** [review]
One tiny change request, otherwise looks good.

File: src/structurizr/api/structurizr_client.py (hunk 1 above)
Comment (yt-ms) [reply src/structurizr/api/structurizr_client.py]: Yep, fair enough.

**Review by Midnighter — APPROVED** [review]
:+1:

**Pull request merged**